Data/Repositories/CandidateRepository.cs

```diff
@@ -1,14 +1,56 @@
-﻿
+﻿using Data.Entities;
+
 namespace Data.Repositories
 {
-    public class CandidateRepository 
+    public class CandidateRepository : BaseRepository<Candidate>
     {
-        public void GetCandidatetById(int id) { }
+        public CandidateRepository(DataContext context) : base(context)
+        {
+
+        }
+
+        public Candidate GetCandidateById(int id) {
+            Candidate result = GetById(id);
+
+            if (result == default) return null;
+
+            return result;
+        }
+
+        public IEnumerable<Candidate> GetAllCandidates() 
+        {
+            IEnumerable<Candidate> result = GetAll();
+
+            return result;
+        }
 
-        public void GetAllCandidate() { }
+        public int AddCandidate(Candidate candidate) {
+            try
+            {
+                AddAndSaveChanges(candidate);
+            }
+            catch (Exception ex)
+            {
+                return -1;
+            }
+            return 1;
+        }
 
-        public void AddCandidate(int id) { }
+        public int UpdateCandidate(Candidate candidate)
```

Comment:
If yoou move try and catch logic to service, you dont need this method because you have UpdateAndSaveChanges